src/main/java/io/confluent/idesidecar/restapi/credentials/ApiKeyAndSecret.java

```diff
@@ -0,0 +1,150 @@
+package io.confluent.idesidecar.restapi.credentials;
+
+import static io.vertx.core.http.HttpHeaders.AUTHORIZATION;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.confluent.idesidecar.restapi.exceptions.Failure;
+import io.confluent.idesidecar.restapi.exceptions.Failure.Error;
+import io.confluent.idesidecar.restapi.models.ConnectionSpec;
+import io.confluent.idesidecar.restapi.models.ConnectionSpec.SecurityProtocol;
+import io.confluent.idesidecar.restapi.models.ConnectionSpec.SslIdentificationAlgorithm;
+import io.quarkus.runtime.annotations.RegisterForReflection;
+import io.vertx.core.MultiMap;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+@Schema(description = "Basic authentication credentials")
+@RegisterForReflection
+public record ApiKeyAndSecret(
+
+    @Schema(description = "The API key to use when connecting to the external service.")
+    @Size(min = 1, max = KEY_MAX_LEN)
+    @JsonProperty(value="api_key")
+    @NotNull
+    String key,
+
+    @Schema(description = "The API secret to use when connecting to the external service.")
+    @Size(min = 1, max = SECRET_MAX_LEN)
+    @JsonProperty(value="api_secret")
+    @NotNull
+    ApiSecret secret
+) implements Credentials {
+
+  private static final int KEY_MAX_LEN = 64;
+  private static final int SECRET_MAX_LEN = 256;
+
+  private static final String PLAIN_LOGIN_MODULE_CLASS =
+      "org.apache.kafka.common.security.plain.PlainLoginModule";
+
+  @Override
+  public Type type() {
+    return Type.API_KEY_AND_SECRET;
+}
+
+  @Override
+  public Optional<Map<String, String>> kafkaClientProperties(
+      KafkaConnectionOptions options
+  ) {
+    var config = new LinkedHashMap<String, String>();
+    var protocol = options.securityProtocol();
+    if (protocol != null && protocol != SecurityProtocol.UNKNOWN) {
+      config.put("security.protocol", protocol.toString());
+    }
+    var algorithm = options.sslIdentificationAlgorithm();
+    if (algorithm != null && algorithm != SslIdentificationAlgorithm.UNKNOWN) {
+      config.put("ssl.endpoint.identification.algorithm", algorithm.toString());
+    }
+    if (protocol != null) {
+      config.put("sasl.mechanism", protocol.isSslEnabled() ? "SASL_SSL" : "PLAIN");
```

Comment:
I'd rather initially try to infer it, because `sasl.mechanism` is mostly driven by the type of credentials (or vice versa). If inferring proves to be awkward or not correct, then we can add it to the connection spec for Kafka in a future PR.

Reply:
Do you know if `SASL_SSL` is a valid value for `sasl.mechanism`? Maybe we could just set it to `PLAIN` for now.

Reply:
Ah, good catch.